### ISSUE TITLE
`yalc publish` error in monorepo, "path" argument must be string

### ISSUE THREAD

**Issue by MitchPierias** [whitecolor/yalc#101]
I'm receiving a "path" error when running `yalc publish --private` from within a monorepo setup.

```
TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received type undefined
```

### Exception

```
(node:5644) UnhandledPromiseRejectionWarning: TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received type undefined
    at validateString (internal/validators.js:125:11)
    at Object.join (path.js:1147:7)
    at Object.<anonymous> (/Users/mitch/.config/yarn/global/node_modules/yalc/src/copy.js:95:47)
    at step (/Users/mitch/.config/yarn/global/node_modules/yalc/src/copy.js:43:23)
    at Object.next (/Users/mitch/.config/yarn/global/node_modules/yalc/src/copy.js:24:53)
    at /Users/mitch/.config/yarn/global/node_modules/yalc/src/copy.js:18:71
    at new Promise (<anonymous>)
    at __awaiter (/Users/mitch/.config/yarn/global/node_modules/yalc/src/copy.js:14:12)
    at Object.exports.copyPackageToStore (/Users/mitch/.config/yarn/global/node_modules/yalc/src/copy.js:87:63)
    at Object.<anonymous> (/Users/mitch/.config/yarn/global/node_modules/yalc/src/publish.js:79:45)
(node:5644) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 2)
(node:5644) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

### Environment
**MacOSX:** 10.14.6
**Node:** 10.16.2
**Yarn:** 1.17.3
**Yalc:** 1.0.0-pre.34

**Comment by whitecolor** [comment]
What is `--private`?

**Comment by MitchPierias** [comment]
@whitecolor `yalc publish` on a monorepo using `yarn workspaces` (packages set to private) outputs

```
Will not publish package with `private: true` use --private flag to force publishing.
```

**Comment by MitchPierias** [comment]
Error was because `version` was not specified in `package.json` which cause the path construction to fail in the `copy` script.

Validation needed for required `version` and `name` fields within the `copyPackageToStore()` function in `copy.ts`

**Comment by slikts** [comment]
I ran into this because the `version` property isn't used when publishing with `semantic-release`.

**Comment by slikts** [comment]
Ran into this again with a different package and had to go through googling again. This issue shouldn't be closed.

**Comment by danielcnascimento** [comment]
I tried to add any version, as mentioned above, but it still doesn't work.

**Comment by samhh** [comment]
As of Yalc 1.0.0-pre.53, in an ordinary repo:

- Without neither name nor verson: `TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received undefined`
- With only name: `TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received undefined`
- With only version: `Package manifest path/to/package.json should contain name and version.`
- With both name and verson: Works.

**Comment by julix-unity** [comment]
Ran in to the same issue with a mono repo context. Had to run `--private` to run it at all - then had the `path` error mentioned above and the solution (adding version) worked as well. That's when I noticed I'm in a mono repo with workspaces, I don't have to `yalc` the whole thing and workspaces have version etc. so I should `yalc` that directly.

`path argument` was not a useful error message for me to realize that (except for finding this thread)...
